### PR TITLE
[CLOUDSTACK-9215]Test to verify vm deployment in vpc tier if nic type is Vmxnet3

### PR DESCRIPTION
**Pull request by sanju1010**
Please check bug CLOUDSTACK-9215 for more details.

Test Results:
==========
Test to create vpc tier with nic type as Vmxnet3 ... === TestName: test_01_create_tier_Vmxnet3 | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 591.630s

OK

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@sanju1010 can you move CS-40430 to the apache jira and refer that one? the ticket in your commit message is not public (to me)

**Comment by sanju1010** [comment]
@DaanHoogland Cloudstack-9215 is the one I created for this PR.

**Comment by DaanHoogland** [comment]
Can you change the commit message with this id, then?

**Comment by sanju1010** [comment]
@DaanHoogland , modified the commit message as per your review comment.

**Comment by DaanHoogland** [comment]
great, thanks
I will try and find a free env to run it.

File: test/integration/component/maint/test_vpc.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+            VirtualMachine.create(
+                self.apiclient,
+                self.services["virtual_machine"],
+                accountid=self.account.name,
+                domainid=self.account.domainid,
+                serviceofferingid=self.service_offering.id,
+                networkids=[str(network.id)]
+            )
+            self.debug("Deployed VM in network: %s" % network.id)
+        except Exception as e:
```
Comment (DaanHoogland) [inline test/integration/component/maint/test_vpc.py]: why rely on the valid implementation by the developer of the error handling in CloudStack? shouldn't you assert that it is actually created? A programmer might have silently ignored an exception.

**Comment by DaanHoogland** [comment]
test succeeds as is : 
```
[root@daans-dev-machinina ~]# ssh cs1 nosetests --with-marvin --marvin-config=/data/shared/marvin/mct-zone1-kvm1-kvm2.cfg  -s -a tags=advanced,required_hardware=true /data/git/cs1/cloudstack/test/integration/component/maint/test_vpc.py
root@cs1's password:

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /tmp//MarvinLogs//Jan_08_2016_14_17_12_EBMCRT. All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
===final results are now copied to: /tmp//MarvinLogs/test_vpc_1F4ULR===
```

LGTM but seems less useful due to my remark on line 354

**Comment by bhaisaab** [comment]
@sanju1010 please squash changes in to a single commit and comment if all outstanding issues are resolved now?

**Comment by bvbharatk** [comment]
_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0

### ACS CI BVT Run
 **Sumarry:**
 Build Number 132
 Hypervisor xenserver
 NetworkType Advanced
 Passed=99
 Failed=5
 Skipped=4


**Failed tests:**
* integration.smoke.test_service_offerings.TestCreateServiceOffering

 * ContextSuite context=TestServiceOfferings>:setup Failed

* integration.smoke.test_loadbalance.TestLoadBalance

 * test_01_create_lb_rule_src_nat Failed

 * test_02_create_lb_rule_non_nat Failed

 * test_assign_and_removal_lb Failed

* integration.smoke.test_volumes.TestCreateVolume

 * test_06_download_detached_volume Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_deploy_vgpu_enabled_vm
test_06_copy_template
test_06_copy_iso

**Passed test suits:**
integration.smoke.test_deploy_vm_with_userdata.TestDeployVmWithUserData
integration.smoke.test_affinity_groups_projects.TestDeployVmWithAffinityGroup
integration.smoke.test_portable_publicip.TestPortablePublicIPAcquire
integration.smoke.test_over_provisioning.TestUpdateOverProvision
integration.smoke.test_global_settings.TestUpdateConfigWithScope
integration.smoke.test_guest_vlan_range.TestDedicateGuestVlanRange
integration.smoke.test_scale_vm.TestScaleVm
integration.smoke.test_routers.TestRouterServices
integration.smoke.test_reset_vm_on_reboot.TestResetVmOnReboot
integration.smoke.test_snapshots.TestSnapshotRootDisk
integration.smoke.test_deploy_vms_with_varied_deploymentplanners.TestDeployVmWithVariedPlanners
integration.smoke.test_network.TestDeleteAccount
integration.smoke.test_non_contigiousvlan.TestUpdatePhysicalNetwork
integration.smoke.test_deploy_vm_iso.TestDeployVMFromISO
integration.smoke.test_public_ip_range.TestDedicatePublicIPRange
integration.smoke.test_multipleips_per_nic.TestDeployVM
integration.smoke.test_regions.TestRegions
integration.smoke.test_affinity_groups.TestDeployVmWithAffinityGroup
integration.smoke.test_network_acl.TestNetworkACL
integration.smoke.test_pvlan.TestPVLAN
integration.smoke.test_ssvm.TestSSVMs
integration.smoke.test_nic.TestNic
integration.smoke.test_deploy_vm_root_resize.TestDeployVM
integration.smoke.test_resource_detail.TestResourceDetail
integration.smoke.test_secondary_storage.TestSecStorageServices
integration.smoke.test_vm_life_cycle.TestDeployVM
integration.smoke.test_disk_offerings.TestCreateDiskOffering

**Comment by sanju1010** [comment]
Squashed changed into single commit and addressed all the review comments and also posted test run results after making changes. Can somebody please merge this PR?

**Comment by swill** [comment]
@DaanHoogland you ran this against real hardware correct?

**Comment by DaanHoogland** [comment]
real virtualized hardware (advanced zone with 2 kvm hosts, bubble style).

**Comment by swill** [comment]
I am not sure how that tests this code though.  I have that same setup, but I don't think I have the ability to test this configuration.  Am I wrong?

**Comment by DaanHoogland** [comment]
No, you are right. I no longer have that 'hardware' so I can rerun the test but as I stated earlier the test was basing itself on an exception not being throw at the time I did the test and therefore not as useful as would be expected. The test has improved but I have no longer access to the  virtualised environment in which I ran it.

However; this is a test, not production code. What do you want from the test other then the test writer being able to use it to verify ACS? Unless testcode can harm production environments I'd say merge @swill @wilderrodrigues @remibergsma @bhaisaab

**Comment by bhaisaab** [comment]
Agree with @DaanHoogland though looks like it ran no tests (skipped them)

File: test/integration/component/maint/test_vpc.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+        """
+            Test to create vpc tier with nic type as Vmxnet3
+            #1.Set global setting parameter "vmware.systemvm.nic.device.type"
+            to "Vmxnet3"
+            #2.Create VPC
+            #3.Create one tier
+            #4.Deploy one guest vm in the tier created in step3
+        """
+        if self.hypervisor.lower() not in ['vmware']:
+            self.skipTest("This test can only run on vmware setup")
```
Comment (bhaisaab) [inline test/integration/component/maint/test_vpc.py]: @DaanHoogland @swill to run this test, you'll need a vmware environment. I read the code, LGTM.

Looking at the test this is testing that a particular nic type works in a VPC, than test VPC itself. Ideally, we would want few tests for VPC that can be run against any of the major 3 hypervisors (xen, vmware, kvm) but it's a good start.

**Comment by swill** [comment]
Since this is a net new test that I do not have the ability to test and because the tests that have been run are passing, I think this PR is ready.  Are we in agreement on this?

**Pull request merged**